src/AddNodeToggleSwitch.js

```diff
@@ -0,0 +1,38 @@
+// @flow
+
+import React from 'react';
+import { injectIntl } from 'react-intl';
+import { ReactComponent as AddIcon } from './svg/Add.svg';
+import './AddNodeToggleSwitch.scss';
+
+type AddNodeToggleSwitchProps = {
+    intl: any,
+    isAddNodeExpandedMode: boolean,
+    onClick: () => void,
+    onKeyDown: (e: SyntheticKeyboardEvent<HTMLInputElement>) => void
```

Comment:
I suggest moving the click and keyboard handling from `ProgramBlockEditor` into `AddNodeToggleSwitch` and instead providing an `onChange` callback property. The user actions for toggling the control state are good candidates for being the responsibility of the AddNodeToggleSwitch control. This would enable us to move some code out of `ProgramBlockEditor` and put it closer to the UI for the control and in the future would make it easier to re-use the toggle control for other situations in the UI. Locating the user action logic in the component will ensure that all toggle switches operate the same and that we don't duplicate logic unnecessarily.